### ISSUE TITLE
Prettify user symbols

### ISSUE THREAD

**Issue by ecorm** [KDAB/hotspot#227]
**Is your feature request related to a problem? Please describe.**
When profiling template-heavy code, the symbol names are almost always clipped by the flame graph boxes. It makes it difficult to identify the hotspots at a glance, and one has to hover over each box to identify the function.

The View > Prettify Symbols feature only seems to work with standard library classes.

**Describe the solution you'd like**
I would like a prettify feature that collapses the template parameter list of my own classes to `<...>`. Ideally, the depth of the collapsing would be configurable.

**Describe alternatives you've considered**
I checked for `perf record` options that could do this, but couldn't find any.

**Additional context**
None

**Comment by raphinesse** [comment]
In my fork I implemented [a very simple version of this](https://github.com/raphinesse/hotspot/commit/4732571211f49ff723911a745b0d3616b9aa1632). I just replaced the entire existing prettification with logic that bluntly strips any template arguments at the top level. Very primitive, but it made working with the flame graph much more enjoyable for me.

My approach could easily be extended to support a depth parameter. But I'm not entirely sure how helpful a global depth for all templates would be.

After having worked with my hacked version for a bit, I think custom symbol name transformations would have the potential to further improve the situation. A relatively easy approach would be a user-configurable list of regex/replacement patterns. This could be used to replace long, common symbol names with short aliases: `boost::container::dtl::static_storage_allocator<>` → `bcd::ssa<>`.
